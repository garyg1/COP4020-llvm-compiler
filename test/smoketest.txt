extern int print_int(int x);
extern int print_float(float x);
extern float one_float();
extern int one();

def int k_int() (
    3
);

def float k_float() (
    3.0
);

def int add_int_but_replace_first(int a, int b) (
    a = k_int():
    a + b
);

def float add_float(float a, float b) (
    a + b + one_float() + k_float()
);

def int main() (
    print_int(add_int_but_replace_first(1, 2)):
    print_float(add_float(1.0, 2.0))
);