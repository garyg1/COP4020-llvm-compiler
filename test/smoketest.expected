5
7.000000
